### ISSUE TITLE
[plejd-api] Unable to retrieve session token response:  Request failed with status code 403 Error: Request failed with status code 403

### ISSUE THREAD

**Issue by Rilleboy** [icanos/hassio-plejd#162]
I have just installed my raspberry pi4, and the purpose of it was that i would like to have all my gadgets in the same box.
but i cant get the Plejd integration to work..

this is what i get when i try to start it up...

2021-02-10 00:02:41 ERR [plejd-api] Unable to retrieve session token response:  Request failed with status code 403
Error: Request failed with status code 403
    at createError (/plejd/node_modules/axios/lib/core/createError.js:16:15)
    at settle (/plejd/node_modules/axios/lib/core/settle.js:17:12)
    at IncomingMessage.handleStreamEnd (/plejd/node_modules/axios/lib/adapters/http.js:260:11)
    at IncomingMessage.emit (events.js:326:22)
    at endReadableNT (_stream_readable.js:1241:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
(node:302) UnhandledPromiseRejectionWarning: Error: unable to retrieve session token response: Error: Request failed with status code 403
    at /plejd/PlejdApi.js:67:18
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:302) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:302) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code

**Comment by SweVictor** [comment]
Status 403 is "Forbidden" meaning credentials are correct (but please double-check), but Plejd server still won't process request. I have been getting those from time to time during testing, after some time (minutes/hours, turn the addon off between retries).

Testing manually I realized that cloud.plejd.com has two IP addresses, I have had the case where one of them worked and the other not. I don't know of any way to force HA to use the "secondary" IP address though.

Anyway, for me the error has gone away after a few hours so I haven't looked into it anymore.

**Comment by icanos** [comment]
I'm guessing they use some kind of API gateway which implements throttling of requests or has some abuse detection algorithms which blocks further requests for X minutes if abuse/too many requests are detected. That's why we get that on testing and/or if the addon fails to start and queries the API multiple times.

**Comment by Rilleboy** [comment]
Thanks. i tried to "rebuild" and got this following message instead.

[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[11:47:39] INFO: Starting the Plejd service...
[11:47:39] INFO: Verified permissions on startup script
[11:47:39] INFO: Executing startup script
[11:47:39] INFO: Running add-on
2021-02-10 11:47:40 INF [plejd-main] Log level set to info
2021-02-10 11:47:40 INF [plejd-main] Starting Plejd add-on v. 0.5.1
2021-02-10 11:47:40 INF [plejd-api] login()
2021-02-10 11:47:40 INF [plejd-api] logging into Hem
2021-02-10 11:47:41 INF [plejd-api] got session token response
2021-02-10 11:47:41 INF [plejd-api] Get all Plejd sites for account...
2021-02-10 11:47:41 INF [plejd-api] got site list response
2021-02-10 11:47:41 INF [plejd-api] Get site details...
2021-02-10 11:47:42 INF [plejd-api] got site details response
2021-02-10 11:47:42 INF [plejd-mqtt] Initializing MQTT connection for Plejd addon
2021-02-10 11:47:42 INF [plejd-ble] Starting Plejd BLE, resetting all device states.
2021-02-10 11:47:42 INF [plejd-ble] wireEvents()
2021-02-10 11:47:42 INF [plejd-ble] init()
events.js:291
      throw er; // Unhandled 'error' event
      ^
Error: Not supported
    at Parser._parsePayload (/plejd/node_modules/mqtt-packet/parser.js:128:27)
    at Parser.parse (/plejd/node_modules/mqtt-packet/parser.js:43:45)
    at Writable.writable._write (/plejd/node_modules/mqtt/lib/client.js:293:12)
    at doWrite (/plejd/node_modules/readable-stream/lib/_stream_writable.js:428:64)
    at writeOrBuffer (/plejd/node_modules/readable-stream/lib/_stream_writable.js:417:5)
    at Writable.write (/plejd/node_modules/readable-stream/lib/_stream_writable.js:334:11)
    at Socket.ondata (_stream_readable.js:718:22)
    at Socket.emit (events.js:314:20)
    at addChunk (_stream_readable.js:297:12)
    at readableAddChunk (_stream_readable.js:272:9)
Emitted 'error' event on MqttClient instance at:
    at Parser.emit (events.js:314:20)
    at Parser._emitError (/plejd/node_modules/mqtt-packet/parser.js:685:10)
    at Parser._parsePayload (/plejd/node_modules/mqtt-packet/parser.js:128:16)
    at Parser.parse (/plejd/node_modules/mqtt-packet/parser.js:43:45)
    [... lines matching original stack trace ...]
    at Socket.emit (events.js:314:20)

**Comment by Cederstrom96** [comment]
Same problem here

ERR [plejd-api] Unable to retrieve session token response:  Request failed with status code 403
Error: Request failed with status code 403
    at createError (/plejd/node_modules/axios/lib/core/createError.js:16:15)
    at settle (/plejd/node_modules/axios/lib/core/settle.js:17:12)
    at IncomingMessage.handleStreamEnd (/plejd/node_modules/axios/lib/adapters/http.js:260:11)
    at IncomingMessage.emit (events.js:326:22)
    at endReadableNT (_stream_readable.js:1241:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
(node:33182) UnhandledPromiseRejectionWarning: Error: unable to retrieve session token response: Error: Request failed with status code 403
    at /plejd/PlejdApi.js:67:18
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:33182) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:33182) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by SweVictor** [comment]
@Cederstrom96 : Probably same issue then - requests are throttled by the Plejd API. Stop the addon for a while and try again.

@Rilleboy : This is a completely different error. I have actually never seen exceptions from the mqtt library, so no real clue what happens here. Has 0.5.1 worked before for you? Have you updated anything else?

That the original error re-appears is due to code structure of the 0.5.X (and earlier) releases, I'm trying to address them in a larger code restructure. Basically what happens is that unhandled exceptions will restart the addon, meaning new requests to the Plejd API. If this happens too often, you will get throttled by the Plejd API.

**Comment by Rilleboy** [comment]
This is a new installation, i have made a test before on an old laptop where i installed Debian and Home-assistant. Did not get it to work there either. I found the Plejd controllers but couldn’t change them, all I saw was their status. That is why I wanted to test a raspberry instead... 😢

**Comment by Rilleboy** [comment]
When I look in the log now, this is what I see
![Uploading 744E5A39-3292-44A2-9CE4-29A4D14D6270.png…]()

**Comment by Rilleboy** [comment]
![D094B6C3-BE64-47B9-BC53-2ED5A5709F97](https://user-images.githubusercontent.com/78828421/107565213-b9917d00-6be3-11eb-9df4-0d183ed242c2.png)

this is what I get in the log now 🙄

**Comment by Rilleboy** [comment]
![908D6DD0-3857-4CA7-BBB6-ECF8A6A5FA38](https://user-images.githubusercontent.com/78828421/107566049-a7fca500-6be4-11eb-911b-7d976d6ae7f5.png)

can this be a reason? SSL issues?

**Comment by SweVictor** [comment]
Don't know if SSL is required, I would expect not but don't know.

Have you looked at the "Having issues" section in the README? If you can't find the issue I would suggest turning on verbose logging (or even silly for full API response and ALL logs) and export the logs via a root shell (instructions in the Logs section).
 
https://github.com/icanos/hassio-plejd/blob/master/plejd/README.md

Maybe someone else has ideas.

**Comment by Rilleboy** [comment]
I´ve tried all of the descriptions in the "having issues" section and did not get any better results...

**Comment by Rilleboy** [comment]
This is my latest log......

[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 
-----------------------------------------------------------
 Add-on: Plejd
 Adds support for the Swedish home automation devices from Plejd.
-----------------------------------------------------------
 Add-on version: 0.5.1
 You are running the latest version of this add-on.
parse error: Expected string key before ':' at line 1, column 4
[16:40:48] ERROR: Unknown HTTP error occured
 System:   (armv7 / raspberrypi4)
 Home Assistant Core: 2021.2.3
 Home Assistant Supervisor: 2021.02.6
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[16:40:49] INFO: Starting the Plejd service...
[16:40:49] INFO: Verified permissions on startup script
[16:40:49] INFO: Executing startup script
[16:40:49] INFO: Running add-on
2021-02-11 16:40:50 INF [plejd-main] Log level set to info
2021-02-11 16:40:50 INF [plejd-main] Starting Plejd add-on v. 0.5.1
2021-02-11 16:40:50 INF [plejd-api] login()
2021-02-11 16:40:50 INF [plejd-api] logging into Hem
2021-02-11 16:40:51 INF [plejd-api] got session token response
2021-02-11 16:40:51 INF [plejd-api] Get all Plejd sites for account...
2021-02-11 16:40:51 INF [plejd-api] got site list response
2021-02-11 16:40:51 INF [plejd-api] Get site details...
2021-02-11 16:40:52 INF [plejd-api] got site details response
2021-02-11 16:40:52 INF [plejd-mqtt] Initializing MQTT connection for Plejd addon
2021-02-11 16:40:52 INF [plejd-ble] Starting Plejd BLE, resetting all device states.
2021-02-11 16:40:52 INF [plejd-ble] wireEvents()
2021-02-11 16:40:52 INF [plejd-ble] init()
2021-02-11 16:40:52 INF [plejd-ble] disconnecting /org/bluez/hci0/dev_F8_8D_17_36_9A_09
2021-02-11 16:40:57 INF [plejd-ble] Connecting to /org/bluez/hci0/dev_F8_8D_17_36_9A_09
2021-02-11 16:41:08 INF [plejd-ble] onDeviceConnected()
2021-02-11 16:41:08 INF [plejd-ble] trying 10 characteristics
2021-02-11 16:41:08 INF [plejd-ble] authenticate()
2021-02-11 16:41:08 INF [plejd-ble] startPing()
2021-02-11 16:41:08 INF [plejd-ble] startWriteQueue()
2021-02-11 16:41:57 WRN [plejd-mqtt] Unknown device id 25 - not handled by us.
2021-02-11 16:42:16 WRN [plejd-mqtt] Unknown device id 26 - not handled by us.

**Comment by SweVictor** [comment]
Sorry to hear that. Without turning on verbose logging or silly (shows VRB/DBG logs in the log) it's hard to debug unfortunately.

**Comment by Rilleboy** [comment]
@SweVictor what do I do with the log then?

**Comment by SweVictor** [comment]
Well... If you're quick (or the log stalls) you might be able to copy the startup on verbose level. Otherwise export the file, scrub any data you deem sensitive and upload somewhere - https://pastebin.com/ for example.

**Comment by Rilleboy** [comment]
Like this??

[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 
-----------------------------------------------------------
 Add-on: Plejd
 Adds support for the Swedish home automation devices from Plejd.
-----------------------------------------------------------
 Add-on version: 0.5.1
 You are running the latest version of this add-on.
parse error: Expected string key before ':' at line 1, column 4
[23:36:09] ERROR: Unknown HTTP error occured
 System:   (armv7 / raspberrypi4)
 Home Assistant Core: 2021.2.3
 Home Assistant Supervisor: 2021.02.6
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[23:36:10] INFO: Starting the Plejd service...
[23:36:10] INFO: Verified permissions on startup script
[23:36:10] INFO: Executing startup script
[23:36:10] INFO: Running add-on
2021-02-11 23:36:11 VRB [plejd-main] Log level set to verbose
2021-02-11 23:36:11 INF [plejd-main] Starting Plejd add-on v. 0.5.1
2021-02-11 23:36:11 INF [plejd-api] login()
2021-02-11 23:36:11 INF [plejd-api] logging into Hem
2021-02-11 23:36:11 DBG [plejd-api] sending POST to https://cloud.plejd.com/parse/login
2021-02-11 23:36:12 ERR [plejd-api] Unable to retrieve session token response:  Request failed with status code 403
Error: Request failed with status code 403
    at createError (/plejd/node_modules/axios/lib/core/createError.js:16:15)
    at settle (/plejd/node_modules/axios/lib/core/settle.js:17:12)
    at IncomingMessage.handleStreamEnd (/plejd/node_modules/axios/lib/adapters/http.js:260:11)
    at IncomingMessage.emit (events.js:326:22)
    at endReadableNT (_stream_readable.js:1241:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
(node:302) UnhandledPromiseRejectionWarning: Error: unable to retrieve session token response: Error: Request failed with status code 403
    at /plejd/PlejdApi.js:67:18
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:302) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:302) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by Rilleboy** [comment]
2021-02-12 08:43:49 VRB [plejd-ble] Raw event received: 010110001bc53f266001
2021-02-12 08:43:49 VRB [plejd-ble] Device 1, cmd 001b, state NaN, dim/data2 63
2021-02-12 08:43:52 VRB [plejd-mqtt] Warning: mqtt channel closed event, reconnecting...
2021-02-12 08:44:00 VRB [plejd-ble] Raw event received: 010110001bca3f266001
2021-02-12 08:44:00 VRB [plejd-ble] Device 1, cmd 001b, state NaN, dim/data2 63
2021-02-12 08:44:21 VRB [plejd-ble] Raw event received: 010110001be53f266001
2021-02-12 08:44:21 VRB [plejd-ble] Device 1, cmd 001b, state NaN, dim/data2 63
2021-02-12 08:44:22 VRB [plejd-mqtt] Warning: mqtt channel closed event, reconnecting...
2021-02-12 08:44:23 VRB [plejd-ble] Raw event received: 0c011000980100f600
2021-02-12 08:44:23 VRB [plejd-ble] Device 12, cmd 0098, state 1, dim/data2 0
2021-02-12 08:44:23 DBG [plejd-ble] Tak (12) got state+dim update. S: 1, D: 0
2021-02-12 08:44:23 WRN [plejd-mqtt] Unknown device id 12 - not handled by us.
2021-02-12 08:44:23 VRB [plejd-ble] All states: {"11":{"state":1,"dim":48},"12":{"state":1,"dim":0},"13":{"state":1,"dim":255},"17":{"state":1,"dim":255},"19":{"state":0,"dim":208},"23":{"state":1,"dim":255},"24":{"state":0,"dim":0},"25":{"state":1,"dim":44},"26":{"state":0,"dim":98},"28":{"state":1,"dim":255},"31":{"state":1,"dim":59},"32":{"state":1,"dim":95},"33":{"state":0,"dim":192},"35":{"state":1,"dim":255},"36":{"state":0,"dim":128},"39":{"state":0,"dim":128},"41":{"state":0,"dim":64}}

**Comment by icanos** [comment]
Do you have multiple Plejd sites and you have by any chance logged in to the wrong site in Home Assistant?

**Comment by Rilleboy** [comment]
No. i only have one that is named Hem.

**Comment by Rilleboy** [comment]
what i can see it finds all the devices aswell...

2021-02-12 09:03:39 INF [plejd-mqtt] Initializing MQTT connection for Plejd addon
2021-02-12 09:03:39 INF [plejd-ble] Starting Plejd BLE, resetting all device states.
2021-02-12 09:03:39 DBG [plejd-ble] wiring events and waiting for BLE interface to power up.
2021-02-12 09:03:39 INF [plejd-ble] wireEvents()
2021-02-12 09:03:39 INF [plejd-ble] init()
2021-02-12 09:03:39 DBG [plejd-ble] Found BLE interface 'org.bluez.Adapter1' at /org/bluez/hci0
2021-02-12 09:03:39 INF [plejd-ble] disconnecting /org/bluez/hci0/dev_F8_8D_17_36_9A_09
2021-02-12 09:03:41 DBG [plejd-ble] Found Plejd service on /org/bluez/hci0/dev_C3_12_8A_69_68_5E
2021-02-12 09:03:41 DBG [plejd-ble] Found Plejd service on /org/bluez/hci0/dev_F8_8D_17_36_9A_09
2021-02-12 09:03:41 DBG [plejd-ble] Found Plejd service on /org/bluez/hci0/dev_ED_0A_3B_00_F6_77
2021-02-12 09:03:41 DBG [plejd-ble] Found Plejd service on /org/bluez/hci0/dev_C5_9D_17_6A_FC_49
2021-02-12 09:03:41 DBG [plejd-ble] Found Plejd service on /org/bluez/hci0/dev_E9_A9_32_41_75_4B
2021-02-12 09:03:41 DBG [plejd-ble] Found Plejd service on /org/bluez/hci0/dev_E6_6C_C9_B2_BE_56
2021-02-12 09:03:41 DBG [plejd-ble] Found Plejd service on /org/bluez/hci0/dev_F6_56_83_FA_56_8E
2021-02-12 09:03:41 DBG [plejd-ble] Found Plejd service on /org/bluez/hci0/dev_F4_EE_32_30_05_22
2021-02-12 09:03:41 DBG [plejd-ble] Found Plejd service on /org/bluez/hci0/dev_DF_B1_9E_47_35_E6
2021-02-12 09:03:41 DBG [plejd-ble] Found Plejd service on /org/bluez/hci0/dev_F8_CD_F0_DB_AA_D6
2021-02-12 09:03:41 DBG [plejd-ble] Found Plejd service on /org/bluez/hci0/dev_D4_E4_20_BA_ED_D9
2021-02-12 09:03:42 DBG [plejd-ble] Found Plejd service on /org/bluez/hci0/dev_DE_EA_29_1D_D6_02
2021-02-12 09:03:42 DBG [plejd-ble] Found Plejd service on /org/bluez/hci0/dev_C0_B8_7D_18_BD_F8
2021-02-12 09:03:43 DBG [plejd-ble] Got 13 device(s).
2021-02-12 09:03:43 DBG [plejd-ble] Inspecting /org/bluez/hci0/dev_C3_12_8A_69_68_5E
2021-02-12 09:03:43 DBG [plejd-ble] Discovered /org/bluez/hci0/dev_C3_12_8A_69_68_5E with rssi -90
2021-02-12 09:03:43 DBG [plejd-ble] Inspecting /org/bluez/hci0/dev_F8_8D_17_36_9A_09
2021-02-12 09:03:43 DBG [plejd-ble] Discovered /org/bluez/hci0/dev_F8_8D_17_36_9A_09 with rssi -44
2021-02-12 09:03:43 DBG [plejd-ble] Inspecting /org/bluez/hci0/dev_ED_0A_3B_00_F6_77
2021-02-12 09:03:43 DBG [plejd-ble] Discovered /org/bluez/hci0/dev_ED_0A_3B_00_F6_77 with rssi -83
2021-02-12 09:03:43 DBG [plejd-ble] Inspecting /org/bluez/hci0/dev_C5_9D_17_6A_FC_49
2021-02-12 09:03:43 DBG [plejd-ble] Discovered /org/bluez/hci0/dev_C5_9D_17_6A_FC_49 with rssi -73
2021-02-12 09:03:43 DBG [plejd-ble] Inspecting /org/bluez/hci0/dev_E9_A9_32_41_75_4B
2021-02-12 09:03:43 DBG [plejd-ble] Discovered /org/bluez/hci0/dev_E9_A9_32_41_75_4B with rssi -98
2021-02-12 09:03:43 DBG [plejd-ble] Inspecting /org/bluez/hci0/dev_E6_6C_C9_B2_BE_56
2021-02-12 09:03:43 DBG [plejd-ble] Discovered /org/bluez/hci0/dev_E6_6C_C9_B2_BE_56 with rssi -63
2021-02-12 09:03:43 DBG [plejd-ble] Inspecting /org/bluez/hci0/dev_F6_56_83_FA_56_8E
2021-02-12 09:03:43 DBG [plejd-ble] Discovered /org/bluez/hci0/dev_F6_56_83_FA_56_8E with rssi -76
2021-02-12 09:03:43 DBG [plejd-ble] Inspecting /org/bluez/hci0/dev_F4_EE_32_30_05_22
2021-02-12 09:03:43 DBG [plejd-ble] Discovered /org/bluez/hci0/dev_F4_EE_32_30_05_22 with rssi -83
2021-02-12 09:03:43 DBG [plejd-ble] Inspecting /org/bluez/hci0/dev_DF_B1_9E_47_35_E6
2021-02-12 09:03:43 DBG [plejd-ble] Discovered /org/bluez/hci0/dev_DF_B1_9E_47_35_E6 with rssi -91
2021-02-12 09:03:43 DBG [plejd-ble] Inspecting /org/bluez/hci0/dev_F8_CD_F0_DB_AA_D6
2021-02-12 09:03:43 DBG [plejd-ble] Discovered /org/bluez/hci0/dev_F8_CD_F0_DB_AA_D6 with rssi -75
2021-02-12 09:03:43 DBG [plejd-ble] Inspecting /org/bluez/hci0/dev_D4_E4_20_BA_ED_D9
2021-02-12 09:03:43 DBG [plejd-ble] Discovered /org/bluez/hci0/dev_D4_E4_20_BA_ED_D9 with rssi -80
2021-02-12 09:03:43 DBG [plejd-ble] Inspecting /org/bluez/hci0/dev_DE_EA_29_1D_D6_02
2021-02-12 09:03:43 DBG [plejd-ble] Discovered /org/bluez/hci0/dev_DE_EA_29_1D_D6_02 with rssi -89
2021-02-12 09:03:43 DBG [plejd-ble] Inspecting /org/bluez/hci0/dev_C0_B8_7D_18_BD_F8
2021-02-12 09:03:43 DBG [plejd-ble] Discovered /org/bluez/hci0/dev_C0_B8_7D_18_BD_F8 with rssi -99
2021-02-12 09:03:43 INF [plejd-ble] Connecting to /org/bluez/hci0/dev_F8_8D_17_36_9A_09
2021-02-12 09:03:45 DBG [plejd-ble] Found Plejd service on /org/bluez/hci0/dev_F5_05_CE_E0_8E_3E
2021-02-12 09:03:54 INF [plejd-ble] onDeviceConnected()
2021-02-12 09:03:54 DBG [plejd-ble] Device: [object Object]
2021-02-12 09:03:54 INF [plejd-ble] trying 10 characteristics
2021-02-12 09:03:54 DBG [plejd-ble] found PING characteristic.
2021-02-12 09:03:54 DBG [plejd-ble] found AUTH characteristic.
2021-02-12 09:03:54 DBG [plejd-ble] found LAST_DATA characteristic.
2021-02-12 09:03:54 DBG [plejd-ble] found DATA characteristic.
2021-02-12 09:03:54 INF [plejd-ble] authenticate()
2021-02-12 09:03:54 DBG [plejd-ble] Sending challenge to device
2021-02-12 09:03:54 DBG [plejd-ble] Reading response from device
2021-02-12 09:03:54 DBG [plejd-ble] Responding to authenticate
2021-02-12 09:03:54 INF [plejd-ble] startPing()
2021-02-12 09:03:54 INF [plejd-ble] startWriteQueue()
2021-02-12 09:04:09 VRB [plejd-mqtt] Warning: mqtt channel closed event, reconnecting...
2021-02-12 09:04:24 VRB [plejd-ble] Raw event received: 0c011000980100f600
2021-02-12 09:04:24 VRB [plejd-ble] Device 12, cmd 0098, state 1, dim/data2 0
2021-02-12 09:04:24 DBG [plejd-ble] Tak (12) got state+dim update. S: 1, D: 0
2021-02-12 09:04:24 WRN [plejd-mqtt] Unknown device id 12 - not handled by us.
2021-02-12 09:04:24 VRB [plejd-ble] All states: {"12":{"state":1,"dim":0}}
2021-02-12 09:04:39 VRB [plejd-mqtt] Warning: mqtt channel closed event, reconnecting...

**Comment by icanos** [comment]
There seems to be some misconfiguration of your MQTT server, either in the Plejd addon or the configuration of the actual server.

Err:
2021-02-12 09:04:09 VRB [plejd-mqtt] Warning: mqtt channel closed event, reconnecting...

That's why it doesn't work, HA doesn't fully know your devices yet.

**Comment by Rilleboy** [comment]
I think i´ve tried everything now, starting to feel like a r**tard......

i have uninstalled and reinstalled everytning once again and now i get this error......

 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[15:02:07] INFO: Starting the Plejd service...
[15:02:07] INFO: Verified permissions on startup script
[15:02:08] INFO: Executing startup script
[15:02:08] INFO: Running add-on
2021-02-12 15:02:09 INF [plejd-main] Log level set to info
2021-02-12 15:02:09 INF [plejd-main] Starting Plejd add-on v. 0.5.1
2021-02-12 15:02:09 INF [plejd-api] login()
2021-02-12 15:02:09 INF [plejd-api] logging into Hem
2021-02-12 15:02:10 INF [plejd-api] got session token response
2021-02-12 15:02:10 INF [plejd-api] Get all Plejd sites for account...
2021-02-12 15:02:10 INF [plejd-api] got site list response
2021-02-12 15:02:10 INF [plejd-api] Get site details...
2021-02-12 15:02:12 INF [plejd-api] got site details response
(node:302) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'filter' of undefined
    at PlejdApi.getDevices (/plejd/PlejdApi.js:250:41)
    at /plejd/main.js:42:34
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:302) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:302) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by SweVictor** [comment]
Hmm... Sorry to hear you're still having issues. I presume you have verified that you get the mqtt messages from Plejd in the mqtt addon in HA?

If so, I suggest you
* copy the configuration or write down the values
* uninstall the official plejd addon
* install my latest code restructure build from https://github.com/SweVictor/hassio-plejd/tree/features/code-architecture
* set the correct values in the config (don't copy the entire yaml since values have been added)
* set log level to verbose
* potentially set addon to prefer cached API response during testing to avoid slamming the plejd API and get 403 errors

After installing, logs should look like below (notice version 0.6.0-dev, as well as VRB logs).

```
2021-02-13 12:57:10 VRB [plejd-main] Log level set to verbose
Starting Plejd addon and reading configuration...
2021-02-13 12:57:10 VRB [plejd-main] Log level set to verbose
2021-02-13 12:57:10 INF [plejd-main] Plejd add-on, version 0.6.0-dev
2021-02-13 12:57:10 VRB [plejd-main] Addon info: {"name":"Plejd","version":"0.6.0-dev","slug":"plejd","description":"Adds support for the Swedish home automation devices from Plejd.","url":"https://github.com/icanos/hassio-plejd/","arch":["armhf","armv7","aarch64","amd64","i386"],"startup":"application","boot":"auto","host_network":true,"host_dbus":true,"apparmor":false}
2021-02-13 12:57:10 INF [plejd-ble] Starting Plejd BLE Handler, resetting all device states.
2021-02-13 12:57:10 INF [plejd-main] Main Plejd addon init()...
...
2021-02-11 22:36:14 INF [plejd-mqtt] Initializing MQTT connection for Plejd addon
2021-02-11 22:36:14 INF [plejd-ble] init()
2021-02-11 22:36:14 VRB [plejd-ble] dbus-next connected
2021-02-11 22:36:14 VRB [plejd-ble] Managed paths[
... list of BLE paths
...
2021-02-11 22:36:14 DBG [plejd-ble] Found BLE interface 'org.bluez.Adapter1' at /org/bluez/hci0
2021-02-11 22:36:14 INF [plejd-mqtt] Connected to MQTT.
2021-02-11 22:36:14 VRB [plejd-main] connected to mqtt.
2021-02-11 22:36:15 DBG [plejd-mqtt] Sending discovery of X device(s).
2021-02-11 22:36:15 DBG [plejd-mqtt] Sending discovery for xyz
2021-02-11 22:36:15 INF [plejd-mqtt] Discovered light (DIM-01) named xyz with PID xx.
... discovery of all your devices (this messages you can pick up the HA mqtt addon page)
2021-02-11 22:36:15 VRB [plejd-mqtt] Sent mqtt config command for light, xyz
2021-02-11 22:36:15 VRB [plejd-mqtt] Sent mqtt config command for light, xyz (xx)
...
2021-02-11 22:36:20 INF [plejd-ble] Connected device is a Plejd device with the right characteristics.
2021-02-11 22:36:20 INF [plejd-ble] authenticate()
2021-02-11 22:36:20 DBG [plejd-ble] Sending challenge to device
2021-02-11 22:36:20 DBG [plejd-ble] Reading response from device
2021-02-11 22:36:20 DBG [plejd-ble] Responding to authenticate
2021-02-11 22:36:21 VRB [plejd-ble] Stopping discovery...
2021-02-11 22:36:21 VRB [plejd-ble] Stopped BLE discovery
2021-02-11 22:36:21 INF [plejd-ble] BLE Connected to Fönster vardagsrum
2021-02-11 22:36:21 INF [plejd-main] Bluetooth connected. Plejd BLE up and running!
2021-02-11 22:36:21 INF [plejd-ble] startPing()
2021-02-11 22:36:21 INF [plejd-ble] startWriteQueue()
...
2021-02-11 22:36:25 VRB [plejd-ble] Raw event received: 010110001b60b1256001
```

After these steps (API request, mqtt, discovery, BLE) you should be up and running. If you get errors or warnings during startup (or code get stuck without continuing) we can take it from there). I am quite certain you should not get the UnhandledPromiseRejectionWarnings anymore, but if you do I'd like to know 😃 

Good luck!

**Comment by Rilleboy** [comment]
can you tell me were the Addon folder is located??

**Comment by SweVictor** [comment]
Typically symlinked to / I think. If you're using the ssh addon you might need to enable elevated access in settings. In root console after login you should be good

**Comment by Rilleboy** [comment]
I now hooked up a keyboard and screen to my rpi. enbled elevated acces in terminal. I found the addon folder, but now i cant add a folder there. it says acces denied.. Then i thought that i would "play" around a little with the 0.5.1 version so i installed it again and now it found my 20 Plejd devices.. BUT only 13 of them works, and on the other 7 has a mqtt error.....

I'Ḿ REALLY VERY GRATEFUL FOR YOUR PATIENCE!!

**Comment by SweVictor** [comment]
I use the ssh addon with a setting to enable full access
https://community.home-assistant.io/t/home-assistant-community-add-on-ssh-web-terminal/33820

**Comment by Rilleboy** [comment]
I now finally got it to work. I dont know how i did it tho... I decided to reinstall everything again and finally it started up as i should. I described earlier that i 7 of my divicec had an error, and i found out that the devices can not have the same name in the Plejd app. Once i renamned them so that they all had different names they popped up in HA.

Thank´s a million for all your help. Otherwise i think i would have put the HA back in its box and into the storage...
